### PR TITLE
chore: backend service alb https support

### PR DESCRIPTION
**Pull request by dannyrandall**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/deploy/deploy.go [hunk 1]
```diff
@@ -1197,35 +1229,47 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
...
+}
+
+func validateALBRuntime(in *validateALBRuntimeInput) error {
+	if in.RoutingRule.Alias.IsEmpty() {
+		if in.Env.HasImportedCerts() {
 			return &errSvcWithNoALBAliasDeployingToEnvWithImportedCerts{
-				name:    d.name,
-				envName: d.env.Name,
+				name:    in.ServiceName,
+				envName: in.Env.Name,
```
Comment (iamhopaul123) [inline internal/pkg/cli/deploy/deploy.go]: Blame me but there's seem to be a bug. When http is disabled (or not set for backend svc)

```yaml
http: false
```

We shouldn't error out saying if any cert is imported, `alias` has to be set.

File: internal/pkg/deploy/cloudformation/stack/backend_svc.go [hunk 2]
```diff
@@ -29,36 +30,53 @@ type backendSvcReadParser interface {
...
 		},
-		manifest: mft,
+		manifest: conf.Manifest,
 		parser:   parser,
-	}, nil
+	}
+
+	if conf.Env.HasImportedCerts() {
+		b.certImported = true
+		b.httpsEnabled = true
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/stack/backend_svc.go]: This shouldn't be true if `http` is disabled or empty.

File: internal/pkg/cli/deploy/deploy.go [hunk 3]
```diff
@@ -1197,35 +1229,47 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
...
+		if err := in.AliasCertValidator.ValidateCertAliases(aliases, in.Env.CustomConfig.ImportCertARNs); err != nil {
+			return fmt.Errorf("validate aliases against the imported certificate for env %s: %w", in.Env.Name, err)
 		}
 		return nil
 	}
-	if d.app.Domain != "" {
-		if err := validateAppVersionForAlias(d.app.Name, d.appVersionGetter); err != nil {
-			logAppVersionOutdatedError(aws.StringValue(d.lbMft.Name))
+
+	if in.App.Domain != "" {
```
Comment (efekarakus) [inline internal/pkg/cli/deploy/deploy.go]: Similar to boss's comment, the app's domain I think isn't relevant for backend service's `alias`.

At least for now, we'll start of requiring only [this if-statement](https://github.com/aws/copilot-cli/pull/3583/files?diff=split&w=0#diff-d23f1b40938f2960e3b84a2e5799f6e56df9805461923e4319fe19bb22e87763R1252) where if alias is set then they must have imported certs for backend services

**Review by efekarakus — COMMENTED** [review]
+1 to all of Penghao's feedback :)

File: internal/pkg/deploy/cloudformation/stack/backend_svc.go [hunk 4]
```diff
@@ -29,36 +30,55 @@ type backendSvcReadParser interface {
...
-		parser:   parser,
-	}, nil
+		manifest:   conf.Manifest,
+		parser:     parser,
+		albEnabled: !conf.Manifest.RoutingRule.EmptyOrDisabled(),
+	}
+
+	if conf.Env.HasImportedCerts() {
+		b.certImported = true
+		b.httpsEnabled = b.albEnabled
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/backend_svc.go:78]: nice

**Review by efekarakus — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]
🚢

**Pull request merged**